CRM/Core/BAO/FinancialTrxn.php

```diff
@@ -39,31 +39,27 @@ public static function create($params) {
     $trxn = new CRM_Financial_DAO_FinancialTrxn();
     $trxn->copyValues($params);
 
-    if (isset($params['fee_amount']) && is_numeric($params['fee_amount'])) {
-      if (!isset($params['total_amount'])) {
-        $trxn->fetch();
-        $params['total_amount'] = $trxn->total_amount;
-      }
-      $trxn->net_amount = $params['total_amount'] - $params['fee_amount'];
+    if (!isset($trxn->total_amount)) {
+      $trxn->fetch();
     }
+    $trxn->net_amount = ($trxn->total_amount ?? 0) - ($trxn->fee_amount ?? 0);
```

Comment:
I'm pretty certain that this refactor is basically doing the same thing but in a much simpler way!